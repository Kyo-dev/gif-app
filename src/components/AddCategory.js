import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length >= 3) {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        value={inputValue}
        type="text"
        onChange={handleInputChange}
        placeholder="Search..."
      />
    </form>
  );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}