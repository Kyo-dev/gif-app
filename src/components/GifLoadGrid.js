import React from "react";

export const GifLoadGrid = ({ title, gif }) => {
  return (
    <div className="card animate__animated animate__backInUp">
      <img src={gif} alt={title} />
      <p>{title}</p>
    </div>
  );
};
