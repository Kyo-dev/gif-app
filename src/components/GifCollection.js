import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifLoadGrid } from "./GifLoadGrid";

export const GifCollection = ({ category }) => {
  const {data:gifs, loading} = useFetchGifs(category);
  return (
    <>
      <h3 className= "card animate__animated animate__backInDown">{category}</h3>
      {loading && <p className ="card animate__animated animate__flash">Loading...</p> }
      <div className="card-container">
        {gifs.map((img) => (
          <GifLoadGrid key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
