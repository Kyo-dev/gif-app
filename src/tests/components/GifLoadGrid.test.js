import React from "react";
import { shallow } from "enzyme";
import { GifLoadGrid } from "../../components/GifLoadGrid";

describe("test in <GifLoadGrid />", () => {
  const title = `I'm a title`;
  const gif = `http://localhost.testing.jpg`;
  const wrapper = shallow(<GifLoadGrid title={title} gif={gif} />);

  test("Show component OK", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should showing title", () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
  });

  test('Get img and alt img', () => {
      const img = wrapper.find('img');
      expect(img.prop('src')).toBe(gif);
      expect(img.prop('alt')).toBe(title);

  });
  
  test('class in Div', () => {
      const div = wrapper.find('div').at(0);
      const className = div.prop('className');
      expect(className.includes('animate__backInUp')).toBe(true);
  });
  
});
