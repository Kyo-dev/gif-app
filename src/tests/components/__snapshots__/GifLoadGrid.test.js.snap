// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifLoadGrid /> Show component OK 1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="I'm a title"
    src="http://localhost.testing.jpg"
  />
  <p>
    I'm a title
  </p>
</div>
`;
